addons/account/models/account_invoice.py

```diff
@@ -63,8 +63,9 @@ def _compute_amount(self):
         amount_untaxed_signed = self.amount_untaxed
         if self.currency_id and self.company_id and self.currency_id != self.company_id.currency_id:
             currency_id = self.currency_id
-            amount_total_company_signed = currency_id._convert(self.amount_total, self.company_id.currency_id, self.company_id, self.date_invoice or fields.Date.today())
-            amount_untaxed_signed = currency_id._convert(self.amount_untaxed, self.company_id.currency_id, self.company_id, self.date_invoice or fields.Date.today())
+            rate_date = self._get_currency_rate_date() or fields.Date.today()
```

Comment:
```suggestion
            rate_date = self._get_currency_rate_date() or fields.Date.context_today(self)
```


Actually It should be aligned at each place `fields.Date.today() `or` fields.Date.context_today(self)`

Reply:
No, because this is not a datetime.

Reply:
Take into account that the company amounts in that case are residual and only for avoiding errors on initializations.

Reply:
> No, because this is not a datetime.

Not sure If I am getting it correct, `fields.Date.context_today(self)` gives date based on Timezone and  `fields.Date.context_today(self) ` is also used at https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_invoice.py#L1143

Reply:
Yes, indeed this serves for returning a date, but if you are in other tz, this gives you the day for your timezone if you are in an edge hour. Anyway, I don't think it's relevant here, as this fallback is merely testimonial for avoiding errors in initializations, as date_invoice gets filled by a default value. Also a valid booked invoice will need the other date. I let the Odoo reviewer to decide if he/she wants the other expression.